### PR TITLE
Bug-fix Esoteric chef

### PR DESCRIPTION
**Pull request by t-m-z**
bug-fix: when deleting blank lines the app crashes
refactor: check for deleting blank lines and provide error messages
refactor: add method.types in German language
refactor: renaming "Input" to "Refridgeretor"
provide support for a download button in the trailing title - here I need some more information

### CODE REVIEW

**Comment by t-m-z** [comment]
the exchange-directory contains 4 files with the documenation: pdf/docx/de/en

File: assets/i18n/en.json [hunk 1]
```diff
@@ -386,7 +386,8 @@
   "chef_interpret" : "Interpreter",
   "chef_generate" : "Generator",
   "chef_code" : "Code",
-  "chef_input" : "Input",
+  "chef_input" : "Input of additional ingredients",
+  "chef_fridgerator" : "Fridgerator",
```
Comment (S-Man42) [inline assets/i18n/en.json:390]: Shouldn't it be _Re_fridgerator? Maybe you can add " (Input)"... As I wouldnt know that R. was the Input ;)

Comment (t-m-z) [reply assets/i18n/en.json:390]: yeah - a valid point. Sticking to the naming convention and design principles - Refrigerator ist the most suitable synonym for input. But you are right - this has to be understood by everybody.

**Comment by t-m-z** [comment]
Merge-Cinflicts due to new selection-list solved

**Pull request merged**